docroot/modules/humsci/hs_blocks/src/Plugin/Block/GroupBlock.php

```diff
@@ -180,25 +192,36 @@ protected function buildAdministrativeSection(array &$components) {
       ];
     }
 
+    $route_params = [
+      'section_storage_type' => $section_storage->getStorageType(),
+      'section_storage' => $section_storage->getStorageId(),
+      'delta' => $section_delta,
+      'group' => $this->configuration['machine_name'],
+    ];
+
+    // Build a contextual id that won't generate errors from ajax.
+    // @see _contextual_id_to_links()
+    // Use contextual link data attributes so that layout builder doesn't
+    // disable the link from being clicked.
+    // @see behaviors.layoutBuilderDisableInteractiveElements() in layout-builder.js
+    $contextual_id = $this->configuration['machine_name'] . ':' . http_build_query($route_params) . ':langcode=en';
+    // Generate a token that matches the contextual id.
+    // @see \Drupal\contextual\ContextualController::render()
+    $contextual_token = Crypt::hmacBase64($contextual_id, Settings::getHashSalt() . $this->privateKey);
+
+    $route_attributes = [
+      'class' => ['use-ajax', 'decanter-button--secondary'],
+      'data-dialog-type' => 'dialog',
+      'data-dialog-renderer' => 'off_canvas',
+      'data-contextual-id' => $contextual_id,
+      'data-contextual-token' => $contextual_token,
+    ];
+
+    // Add the "Add" link to the bottom of the group.
     $components['add_link'] = [
       '#type' => 'link',
       '#title' => $this->t('Add Block to Group'),
-      '#url' => Url::fromRoute('hs_blocks.choose_block',
-        [
-          'section_storage_type' => $section_storage->getStorageType(),
-          'section_storage' => $section_storage->getStorageId(),
-          'delta' => $section_delta,
-          'group' => $this->configuration['machine_name'],
-        ],
-        [
-          'attributes' => [
-            'class' => ['use-ajax', 'new-block__link'],
-            'data-dialog-type' => 'dialog',
-            'data-dialog-renderer' => 'off_canvas',
-            'data-contextual-id' => 'add-group-link',
-          ],
-        ]
-      ),
+      '#url' => Url::fromRoute('hs_blocks.choose_block', $route_params, ['attributes' => $route_attributes]),
```

Comment:
Yeeesh, that was a lot for these simple links.

Reply:
you're telling me. :face_with_head_bandage: